R/importSNV.TCGA.R

```diff
@@ -0,0 +1,323 @@
+            tum_all1_ <- read.table(i, header = TRUE, fill = NA, stringsAsFactors = FALSE,
+                                    na.strings = " ")[,12]
+            tum_all2_ <- read.table(i, header = TRUE, fill = NA,  stringsAsFactors = FALSE,
+                                    na.strings = " ")[,13]
+            tum_samp_barcode_ <- read.table(i, header = TRUE, fill = NA,  stringsAsFactors = FALSE,
+                                            na.strings = " ")[,16]
+            tum_sam_uuid_ <- read.table(i, header = TRUE, fill = NA,  stringsAsFactors = FALSE,
+                                        na.strings = " ")[,33]
+
+            # check if variant_class contains valid values from known list
+            classVknown <- c("Frame_Shift_Del", "Frame_Shift_Ins", "In_Frame_Del", "In_Frame_Ins", "Missense_Mutation",
+                        "Nonsense_Mutation", "Silent", "Splice_Site", "Translation_Start_Site", "Nonstop_Mutation",
+                        "3'UTR", "3'Flank", "5'UTR", "5'Flank", "IGR1" , "Intron", "RNA", "Targeted_Region")
+            myVclasses <- unique(var_class_)
+            if (! all(myVclasses %in% classVknown)) {
+                # drop file
+                classValid <- FALSE
+            } else {
+                classValid <- TRUE
+            }
+
+            # check if vairant_type contains valid values from known list
+            typeVknown <- c("SNP", "DNP", "TNP", "ONP", "INS", "DEL", "Consolidated2")
+            myVtypes <- unique(var_type_)
+            if (! all(myVtypes %in% typeVknown)) {
+                # drop file
+                typeValid <- FALSE
+            } else {
+                typeValid <- TRUE
+            }
+
+            # check chromosome value is valid
+            typeChr <- c("chr1","chr2","chr3","chr4","chr5","chr6","chr7","chr8","chr9","chr10","chr11","chr12","chr13",
+                         "chr14","chr15","chr16","chr17","chr18","chr19","chr20","chr21","chr22","chrX","chrY","chrM")
+            myVchr <- unique(chr_)
+            if (! all(myVchr %in% typeChr)) {
+                # drop file
+                chrValid <- FALSE
+            } else {
+                chrValid <- TRUE
+            }
+
+            if (classValid & typeValid & chrValid == FALSE) {
+                # there was an invalid value
+                # report error in file and log
+                # drop file
+                # go to next
+                myNotes <- c(paste(i, " - File was dropped due to invalid attributes in column chr/class/type"))
+                numFilesInvalid <- numFilesInvalid + 1
+            } else {
+                # all tests passed
+                # add data to rSNV
+                fileasDT <- cbind(hugo_ , chr_ , start_, end_, strand_, var_class_,
+                                  var_type_, ref_allele_, tum_all1_, tum_all2_,
+                                  tum_samp_barcode_, tum_sam_uuid_)
+                if (na.rm) {
+                    na.omit(fileasDT)
```

Comment:
```na.omit``` is a function in the data.table package, isn't it? Doesn't it *return* the modified dataframe?
```myDF <- na.omit(mDF)```
Finally: don't you need to be selective about which columns trigger the omission? E.g not all datastes might have UUIDs to begin with ...